clients/python/tests/unit/rest/client_test.py

```diff
@@ -0,0 +1,235 @@
+        self.assertNotEqual(req1.sent_time.nanos, req2.sent_time.nanos)
+        self.assertNotEqual(req1.sent_time.seconds, req2.sent_time.seconds)
+
+    def test_client_send_success_json(self):
+        session_mock = mock.Mock()
+        post = mock.MagicMock()
+        session_mock.post = post
+        post.return_value = get_stub_response_json()
+        event_arr = [get_stub_event_payload_json()]
+        req = SendEventRequest()
+        expected_req = SendEventRequest()
+        expected_req.req_guid = get_static_uuid()
+        req.req_guid = get_static_uuid()
+        time_in_ns = time.time_ns()
+        req.sent_time.FromNanoseconds(time_in_ns)
+        expected_req.sent_time.FromNanoseconds(time_in_ns)
+        with patch("rest.client.requests.session", return_value=session_mock):
+            rest_client = self._get_rest_client()
+            expected_req.events.append(rest_client._convert_to_event_pb(get_stub_event_payload_json()))
+            serialised_data = json_format.MessageToJson(expected_req)
+            rest_client._get_stub_request = mock.MagicMock()
+            rest_client._get_stub_request.return_value = req
+            rest_client._parse_response = mock.MagicMock()
+            rest_client.send(event_arr)
+            post.assert_called_once_with(url=self.sample_url, data=serialised_data,
+                                         headers={"Content-Type": "application/json"}, timeout=2.0)
+            rest_client._parse_response.assert_called_once_with(post.return_value)
+
+    def test_client_send_success_protobuf(self):
+        session_mock = mock.Mock()
+        post = mock.MagicMock()
+        session_mock.post = post
+        post.return_value = get_stub_response_protobuf()
+        event_arr = [get_stub_event_payload_protobuf()]
+        req = SendEventRequest()
+        expected_req = SendEventRequest()
+        expected_req.req_guid = get_static_uuid()
+        req.req_guid = get_static_uuid()
+        time_in_ns = time.time_ns()
+        req.sent_time.FromNanoseconds(time_in_ns)
+        expected_req.sent_time.FromNanoseconds(time_in_ns)
+        with patch("rest.client.requests.session", return_value=session_mock):
+            rest_client = self._get_rest_client(serialiser=Serialiser.PROTOBUF, wire_type=WireType.PROTOBUF)
+            expected_req.events.append(rest_client._convert_to_event_pb(get_stub_event_payload_protobuf()))
+            serialised_data = expected_req.SerializeToString()
+            rest_client._get_stub_request = mock.MagicMock()
+            rest_client._get_stub_request.return_value = req
+            rest_client._parse_response = mock.MagicMock()
+            rest_client.send(event_arr)
+            post.assert_called_once_with(url=self.sample_url, data=serialised_data,
+                                         headers={"Content-Type": "application/proto"}, timeout=2.0)
+            rest_client._parse_response.assert_called_once_with(post.return_value)
+
```

Comment:
can we also add tests for serializer: JSON, wire: Protobuf and vice versa

Reply:
Ack

Reply:
added